aquadoggo/src/db/stores/document.rs

```diff
@@ -0,0 +1,652 @@
+    use p2panda_rs::test_utils::constants::TEST_SCHEMA_ID;
+
+    use crate::db::stores::document::{DocumentStore, DocumentView};
+    use crate::db::stores::entry::StorageEntry;
+    use crate::db::stores::test_utils::{test_create_operation, test_db};
+
+    fn entries_to_document_views(entries: &[StorageEntry]) -> Vec<DocumentView> {
+        let mut document_views = Vec::new();
+        let mut current_document_view_fields = DocumentViewFields::new();
+        for entry in entries {
+            let operation_id: OperationId = entry.hash().into();
+            for (name, value) in entry.operation().fields().unwrap().iter() {
+                if entry.operation().is_delete() {
+                    continue;
+                } else {
+                    current_document_view_fields
+                        .insert(name, DocumentViewValue::new(&operation_id, value));
+                }
+            }
+            let document_view_fields = DocumentViewFields::new_from_operation_fields(
+                &operation_id,
+                &entry.operation().fields().unwrap(),
+            );
+            let document_view =
+                DocumentView::new(&operation_id.clone().into(), &document_view_fields);
+            document_views.push(document_view)
+        }
+        document_views
+    }
+
+    #[tokio::test]
+    async fn inserts_gets_one_document_view() {
+        let (storage_provider, key_pairs, _documents) = test_db(1, 1, false).await;
+        let key_pair = key_pairs.get(0).unwrap();
+        let author = Author::try_from(key_pair.public_key().to_owned()).unwrap();
+
+        // Get one entry from the pre-polulated db
+        let entry = storage_provider
+            .get_entry_at_seq_num(&author, &LogId::new(1), &SeqNum::new(1).unwrap())
+            .await
+            .unwrap()
+            .unwrap();
+
+        // Construct a `DocumentView`
+        let operation_id: OperationId = entry.hash().into();
+        let document_view_id: DocumentViewId = operation_id.clone().into();
+        let document_view = DocumentView::new(
+            &document_view_id,
+            &DocumentViewFields::new_from_operation_fields(
+                &operation_id,
+                &entry.operation().fields().unwrap(),
+            ),
+        );
+
+        // Insert into db
+        let result = storage_provider
+            .insert_document_view(&document_view, &SchemaId::from_str(TEST_SCHEMA_ID).unwrap())
+            .await;
+
+        assert!(result.is_ok());
+
+        let retrieved_document_view = storage_provider
+            .get_document_view_by_id(&document_view_id)
+            .await
+            .unwrap()
+            .unwrap();
+
+        for key in [
+            "username",
+            "age",
+            "height",
+            "is_admin",
+            "profile_picture",
+            "many_profile_pictures",
+            "special_profile_picture",
+            "many_special_profile_pictures",
```

Comment:
Woah, this application must really need many profile pictures 😛

Reply:
haha, indeed, suggestions for better field names welcome ;-p

Reply:
:smile_cat: Not really! Maybe `my_cats_profile_pictures`, `my_turtles_profile_pictures`, `my_birds_profile_pictures`?